bpf/bpf_sock.c

```diff
@@ -329,6 +329,11 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	bool backend_from_affinity = false;
 	__u32 backend_id = 0;
 
+#ifdef SKIP_SK_LB_IN_POD_NS
+	if (!in_hostns)
+		return -EPERM;
+#endif
```

Comment:
nit: would slighly adapt into the below to reduce ifdefs. compiler will optimize out the other condition anyway:
```
  [...]
        if (is_defined(ENABLE_HOST_XLATE_ONLY) && !in_hostns)
                return -ENXIO;
        if (!udp_only && !sock_proto_enabled(ctx->protocol))
                return -ENOTSUP;
  [...]
```

Reply:
Curious, why return `-ENXIO`?

Reply:
Shouldn't matter much but it's consistent to `if (!svc)` case where we have no service.

Reply:
I see, done :)